### PR TITLE
Feature/cluster templates

### PR DESCRIPTION
**Pull request by piontec**
Tempalting values are still broken. 
Encryption keys are in place, just nothing is encrypted yet.

### CODE REVIEW

File: bases/app_sets/hello-web-app/kustomization.yaml [hunk 1]
```diff
@@ -7,11 +7,11 @@ configMapGenerator:
   - behavior: replace
     files:
     - values=default_config_simple_db.yaml
-    name: simple-db-values # has to be in sync with the name used by included app
+    name: ${workload_cluster_name}-simple-db-values # has to be in sync with the name used by included app
```
Comment (MarcelMue) [inline bases/app_sets/hello-web-app/kustomization.yaml:10]: Why not use `cluster_id` everywhere? This seems extremely inconsistent to me now - especially considering the value of the variable will be the same essentially everywhere?

File: management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/FROM_TEMPLATE_WC_NAME/FROM_TEMPLATE_WC_NAME.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
```
Comment (ljakimczuk) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/FROM_TEMPLATE_WC_NAME/FROM_TEMPLATE_WC_NAME.yaml:2]: Let's move it one directory up, see the structure, this is already managed by the `-gitops` Kustomization CR.

Comment (ljakimczuk) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/FROM_TEMPLATE_WC_NAME/FROM_TEMPLATE_WC_NAME.yaml:2]: Let's move it one directory up, see the structure, this is already managed by the `-gitops` Kustomization CR.

File: management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/WC_NAME.yaml [hunk 3]
```diff
@@ -4,6 +4,10 @@ metadata:
   name: ${workload_cluster_name}
```
Comment (ljakimczuk) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/WC_NAME.yaml:4]: same applies here, this file is already present in the parent directory, no need to keep it in the `WC_NAME`

Comment (ljakimczuk) [inline management-clusters/MC_NAME/organizations/ORG_NAME/workload-clusters/WC_NAME/WC_NAME.yaml:4]: same applies here, this file is already present in the parent directory, no need to keep it in the `WC_NAME`

File: bases/app_sets/hello-web-app/kustomization.yaml (hunk 1 above)
Comment (piontec) [reply bases/app_sets/hello-web-app/kustomization.yaml:10]: Like I mentioned above - that's a total mess, not only in this PR. I'll create one another to clean this mess out, as it's wrong in other parts of the repo as well.

File: bases/cluster_templates/dev_cluster_small/cluster/patch_awsmachinedeployment.yaml [hunk 4]
```diff
@@ -0,0 +1,19 @@
+
```
Comment (MarcelMue) [inline bases/cluster_templates/dev_cluster_small/cluster/patch_awsmachinedeployment.yaml:1]:
```suggestion
```

File: bases/cluster_templates/dev_cluster_small/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+
+# TODO: delete this file! Dev time file only!
```
Comment (MarcelMue) [inline bases/cluster_templates/dev_cluster_small/kustomization.yaml:2]: Should this be removed? What's going on here?

Comment (piontec) [reply bases/cluster_templates/dev_cluster_small/kustomization.yaml]: Thanks, forgot about it.

**Review by MarcelMue — APPROVED** [review]

File: bases/cluster_templates/dev_cluster_small/app_sets/hello-web-app-1/kustomization.yaml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+buildMetadata: [originAnnotations]
+configMapGenerator:
+  - behavior: replace
+    files:
+      - values=override_config_hello_world.yaml
+    name: ${workload_cluster_name}-hello-world-values
+generatorOptions:
+  disableNameSuffixHash: true
+kind: Kustomization
```
Comment (uvegla) [inline bases/cluster_templates/dev_cluster_small/app_sets/hello-web-app-1/kustomization.yaml:10]: If it makes sense, I would pull up `kind` after `apiVersion` in these resources as well.

**Review by uvegla — APPROVED** [review]
LGTM from what I can tell. 👍🏻

**Pull request merged**